includes/sanitizers/class-amp-block-sanitizer.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class AMP_Block_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Block_Sanitizer
+ *
+ * Modifies elements created as blocks to match the blocks' AMP-specific configuration.
+ */
+class AMP_Block_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Tag.
+	 *
+	 * @var string Figure tag to identify wrapper around AMP elements.
+	 * @since 1.0
+	 */
+	public static $tag = 'figure';
+
+	/**
+	 * AMP elements.
+	 *
+	 * @todo Add all elements.
+	 * @var array AMP elements to add layout to.
+	 */
+	public static $amp_embeds = array(
+		'amp-youtube',
+		'amp-facebook',
+		'amp-twitter',
+		'amp-vimeo',
+		'amp-instagram',
+		'amp-dailymotion',
+		'amp-hulu',
+		'amp-reddit',
+		'amp-soundcloud',
```

Comment:
Instead of enumerating out all of the embeds, what about this: look for `figure` elements that contain the `wp-block-embed` class, and then if this `figure` element contains a `DOMElement` whose name begins with `amp-` then assume it is an AMP embed component. This will help prevent us having to maintain this list here in response to Gutenberg changes and spec changes.